react-responsive-modal/src/FocusTrap.tsx

```diff
@@ -6,11 +6,16 @@ import {
   getAllTabbingElements,
 } from './lib/focusTrapJs';
 
+export interface FocusTrapOptions {
+  focusOn?: 'firstFocusableElement' | 'modalRoot';
```

Comment:
I am not sure about this approach to be honest, wouldn't be better to allow any react ref as a prop? That way you could really select which element you want to select. If you want to select the 2nd input it would be easy by just passing the react ref of the element. What do you think?

Reply:
Your idea seems great, I implemented it to see how it looks to you. It has one downside: we need to focus on the next animation frame, because we need to wait for refs to propagate to support my use-case: focusing on the modal root.